server/mldb_server.cc

```diff
@@ -410,24 +411,37 @@ scanPlugins(const std::string & dir_)
         {
             try {
                 auto manifest = jsonDecodeStream<PluginManifest>(stream);
+                if (manifest.config.type == "sharedLibrary") {
+                    auto shlibConfig = manifest.config.params.convert<SharedLibraryConfig>();
+                    // strip off the file:// prefix
+                    shlibConfig.address = string(dir, 7);
+                    shlibConfig.allowInsecureLoading = true;
 
-                auto shlibConfig = manifest.config.params.convert<SharedLibraryConfig>();
-                // strip off the file:// prefix
-                shlibConfig.address = string(dir, 7);
-                shlibConfig.allowInsecureLoading = true;
+                    manifest.config.params = shlibConfig;
 
-                manifest.config.params = shlibConfig;
-
-                auto plugin = plugins->obtainEntitySync(manifest.config,
-                                                        nullptr /* on progress */);
+                    auto plugin = plugins->obtainEntitySync(
+                        manifest.config, nullptr /* on progress */);
+                }
+                else if (manifest.config.type == "python" ||
+                         manifest.config.type == "javascript") {
+                    auto config = manifest.config.params.convert<PluginResource>();
+                    config.address = dir;
+                    manifest.config.params = config;
+                    auto plugin = plugins->obtainEntitySync(
+                        manifest.config, nullptr /* on progress */);
+                }
+                else {
+                    throw HttpReturnException(
+                        500, "unknown plugin type to autoload at " + dir);
```

Comment:
Can really throw an HttpReturnException here?  There are no one to catch the response!  I suggest we log an error message instead and continue the processing.